examples/get-started/pure-js/carto/app.js

```diff
@@ -1,98 +1,22 @@
 import mapboxgl from 'mapbox-gl';
 import {Deck} from '@deck.gl/core';
-import {
-  CartoLayer,
-  setDefaultCredentials,
-  BASEMAP,
-  colorBins,
-  API_VERSIONS,
-  MAP_TYPES
-} from '@deck.gl/carto';
-
-const INITIAL_VIEW_STATE = {
-  latitude: 0,
-  longitude: 0,
-  zoom: 1
-};
-
-setDefaultCredentials({
-  apiVersion: API_VERSIONS.V2,
-  username: 'public',
-  apiKey: 'default_public'
-});
-
-// Add Mapbox GL for the basemap. It's not a requirement if you don't need a basemap.
-const map = new mapboxgl.Map({
-  container: 'map',
-  style: BASEMAP.VOYAGER,
-  interactive: false,
-  center: [INITIAL_VIEW_STATE.longitude, INITIAL_VIEW_STATE.latitude],
-  zoom: INITIAL_VIEW_STATE.zoom
-});
-
-// Set the default visible layer
-let visibleLayer = 'airports';
-
-// Create Deck.GL map
-export const deck = new Deck({
-  canvas: 'deck-canvas',
-  width: '100%',
-  height: '100%',
-  initialViewState: INITIAL_VIEW_STATE,
-  controller: true,
-  onViewStateChange: ({viewState}) => {
-    // Synchronize Deck.gl view with Mapbox
-    map.jumpTo({
-      center: [viewState.longitude, viewState.latitude],
-      zoom: viewState.zoom,
-      bearing: viewState.bearing,
-      pitch: viewState.pitch
-    });
-  }
-});
-
-// Add event listener to radio buttons
-document.getElementsByName('layer-visibility').forEach(e => {
-  e.addEventListener('click', () => {
-    visibleLayer = e.value;
-    render();
+import {fetchMap} from '@deck.gl/carto';
+
+const cartoMapId = 'ff6ac53f-741a-49fb-b615-d040bc5a96b8';
+const deck = new Deck({canvas: 'deck-canvas'});
+
+// Get map info from CARTO and update deck
+fetchMap({cartoMapId}).then(({initialViewState, mapStyle, layers}) => {
+  deck.setProps({initialViewState, layers});
```

Comment:
You should set `controller: true` here. The comment below implies that the next `setProps` call is optional, but this statement alone does not enable interaction.